### PR TITLE
feat(accessibility): add focus accessibility action support

### PR DESCRIPTION
**Pull request by pierpo**
## Summary:

I am working on a spatial navigation library, [react-tv-space-navigation](https://github.com/bamlab/react-tv-space-navigation). You can see this lib as the "Flutter of spatial navigation", as it completely bypasses the native focus to implement our own navigation logic in JS. The benefit being the same as Flutter: it is fully cross-platform with a fully predictible behaviour on all platforms.

Our problem (similar to what Flutter had to struggle with, once again 😁 ): the accessibility support is poor.
If you enable TalkBack on Android, our focus system breaks. TalkBack takes over, and the user is stuck with elements that don't move anymore.

The workaround I could think of was to detect the accessibillity focus changes, and propagate them to our lib.

Problem: an event is missing. I opened a PR in React Native but I have no answer. https://github.com/facebook/react-native/pull/41336
Still, this might be a problem for developers working on React Native TV.

As the patch is very simple, I thought we might integrate it here directly.

## Changelog:

[ANDROID] [ADDED] - add accessibilityFocus action support

## Test Plan:

See https://github.com/facebook/react-native/pull/41336

### CODE REVIEW

**Comment by douglowder** [comment]
@pierpo this change seems quite reasonable to me. Sorry for the delay, I have been ill.

@yusufyildirim should also review this.

**Review by douglowder — APPROVED** [review]

**Comment by pierpo** [comment]
> @pierpo this change seems quite reasonable to me. Sorry for the delay, I have been ill.

Thank you so much! So sorry for being pushy 😱
I hope you are doing better 👍

**Comment by robingullo** [comment]
@douglowder If it's not too much to ask, could you also release this a 0.72.6-2 patch ? That would be a great help for my team 😃

**Review by douglowder — CHANGES_REQUESTED** [review]
See comments inline

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java [hunk 1]
```diff
@@ -76,6 +76,7 @@ public class ReactAccessibilityDelegate extends ExploreByTouchHelper {
     sActionIdMap.put("decrement", AccessibilityActionCompat.ACTION_SCROLL_BACKWARD.getId());
     sActionIdMap.put("expand", AccessibilityActionCompat.ACTION_EXPAND.getId());
     sActionIdMap.put("collapse", AccessibilityActionCompat.ACTION_COLLAPSE.getId());
+    sActionIdMap.put("accessibilityFocus", AccessibilityActionCompat.ACTION_ACCESSIBILITY_FOCUS.getId());
```
[the remark this answers is not in the document]
Comment (douglowder) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java:79]: @pierpo can you add support for blur events?

[the remark this answers is not in the document]
Comment (pierpo) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java:79]: Sure, I'll have a look! Many thanks to both of you for the review :)
I'll try this on TextInputs as well!

**Comment by pierpo** [comment]
It works on TextInput 😊 

https://github.com/react-native-tvos/react-native-tvos/assets/6313316/ead278db-777e-4122-a6c5-5275658e08ba

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pierpo) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java:79]: I tried the `ACTION_CLEAR_ACCESSIBILITY_FOCUS` but it never triggers 😢 
I'm not sure it works the way we think. Doc isn't very explicit about it.

I can't find any other event that would make sense as an accessibility blur 🤔

[the remark this answers is not in the document]
Comment (yusufyildirim) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java:79]: Thanks for looking into it! I'll test the PR whenever I can.

[the remark this answers is not in the document]
Comment (pierpo) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java:79]: Hey @yusufyildirim! Any news on the testing? 😁 
Thank you! 🙏

[the remark this answers is not in the document]
Comment (pierpo) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java:79]: Hello 😁 Bumping again! If you ever have the time to test it, we'd be super grateful 😊 
Thanks 🙏

**Review by yusufyildirim — APPROVED** [review]
Amazing stuff! I tested it and it works. Added a little bit of documentation to the README myself. Thank you for adding this and sorry for the long delay 🚀

**Comment by yusufyildirim** [comment]
Let's not forget to cherrypick this for 0.74 and 0.75

**Comment by pierpo** [comment]
Thank you! ❤️ That's awesome 😁

**Review by douglowder — APPROVED** [review]

**Comment by douglowder** [comment]
@yusufyildirim @pierpo this looks great! Go ahead and merge, and I will cherry pick to 0.74 and get a release published.

**Pull request merged**